src/Compilers/CSharp/Test/Symbol/Symbols/EnumTests.cs

```diff
@@ -399,5 +399,17 @@ static void Main()
                 Diagnostic(ErrorCode.ERR_NoTypeDef, "F").WithArguments("A", "UseSiteError_sourceA, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null").WithLocation(5, 20)
                 );
         }
+
+        [Fact]
+        public void PartialPublicEnum()
+        {
+            CreateCompilation("partial public enum E { }").VerifyDiagnostics(
+                // (1,1): error CS0267: The 'partial' modifier can only appear immediately before 'class', 'record', 'struct', 'interface', or a method return type.
+                // partial public enum E { }
+                Diagnostic(ErrorCode.ERR_PartialMisplaced, "partial").WithLocation(1, 1),
+                // (1,21): error CS0267: The 'partial' modifier can only appear immediately before 'class', 'record', 'struct', 'interface', or a method return type.
+                // partial public enum E { }
+                Diagnostic(ErrorCode.ERR_PartialMisplaced, "E").WithLocation(1, 21));
```

Comment:
The duplicate diagnostic is an existing behavior. But whether you want me to fix that in this PR or in a follow-up, I'm okay with that.